ext/oci8/oci8.c

```diff
@@ -1096,9 +1096,7 @@ php_oci_connection *php_oci_do_connect_ex(char *username, int username_len, char
 								}
 
 								if ((tmp_val != NULL) && (tmp != NULL) &&
-									(ZSTR_LEN(tmp->hash_key) == ZSTR_LEN(hashed_details.s)) &&
-									(memcmp(ZSTR_VAL(tmp->hash_key), ZSTR_VAL(hashed_details.s),
-									 ZSTR_LEN(tmp->hash_key)) == 0)) {
+									zend_string_equal_content(tmp->hash_key, hashed_details.s)) {
```

Comment:
Probably should be ``zend_string_equals()``

Reply:
I tried to guess the best names of my new functions based on existing function names - I'll amend my patch tomorrow adjusting to your suggestion.

Reply:
Oh, this was not about my choice of naming, but about my choice of replacement function.
I chose `zend_string_equal_content()` because it does exactly what the code it replaces did: compare lengths, and then `memcmp()` on the value:
https://github.com/php/php-src/blob/dd81dd92643e98ea61bdc9091a47c5cae0925a4b/Zend/zend_string.h#L353-L356
`zend_string_equals()`, on the other hand, additionally checks the `zend_string` pointers for equality:
https://github.com/php/php-src/blob/dd81dd92643e98ea61bdc9091a47c5cae0925a4b/Zend/zend_string.h#L358-L361
This is slightly different from the existing old code. No semantic change, because the outcome of the check is the same. But I wanted to be careful and not introduce actual code changes in this pure-refactoring patch.
Please confirm that you do want this change, and I'll amend the patch.

Reply:
zend_string_equal_content() is a low-level function that is used in some optimized codepaths that have performed a pointer equality check separately. Generally, code should always be using zend_string_equals() unless there's some special reason not to.